lib/Frontend/CompilerInvocation.cpp

```diff
@@ -438,18 +438,27 @@ static bool ParseFrontendArgs(FrontendOptions &Opts, ArgList &Args,
       return true;
     }
   } else if (Opts.RequestedAction != FrontendOptions::NoneAction) {
-    if (Opts.InputFilenames.empty()) {
+    if (Opts.InputFilenames.empty() && !Args.hasArg(OPT_e)) {
       Diags.diagnose(SourceLoc(), diag::error_mode_requires_an_input_file);
       return true;
     }
   }
 
   if (Opts.RequestedAction == FrontendOptions::Immediate) {
-    assert(!Opts.InputFilenames.empty());
-    Opts.ImmediateArgv.push_back(Opts.InputFilenames[0]); // argv[0]
-    if (const Arg *A = Args.getLastArg(OPT__DASH_DASH)) {
-      for (unsigned i = 0, e = A->getNumValues(); i != e; ++i) {
-        Opts.ImmediateArgv.push_back(A->getValue(i));
+    if (Args.hasArg(OPT_e)) {
+      llvm::SmallString<64> CodeBuffer;
+      for (const Arg *A : Args.filtered(OPT_e)) {
+        CodeBuffer.append(A->getValue());
+        CodeBuffer.append("\n");
+      }
+      // SR-5860 WIP TODO: copy CodeBuffer into ImmediateExecutionBuffer
```

Comment:
I'm just guessing here with the `<64>` as the internal length; should it be more? less?

Reply:
Anything power-of-two-y works well for the bounds here.